internal/server/storage/drivers/driver_zfs.go

```diff
@@ -242,8 +249,12 @@ func (d *zfs) Create() error {
 		return err
 	}
 
+	poolType, devices := d.parseSource()
+	if poolType != "" && !slices.Contains(zfsSupportedPoolTypes, poolType) {
+		return fmt.Errorf("Unsupported pool type: %q. Supported pool types: %s", poolType, zfsSupportedPoolTypes)
+	}
 	loopPath := loopFilePath(d.name)
-	if d.config["source"] == "" || d.config["source"] == loopPath {
+	if len(devices) == 1 && devices[0] == loopPath {
```

Comment:
Those two conditions aren't the same, it should likely be:
```
if len(devices) == 0 || (len(devices) == 1 && devices[0] == loopPath)
```

Reply:
At the start of this function (`Create`), `FillConfig` is called, which ensures that `devices` isn't empty. Could we rely on that behavior or check within `Create` as well?

Reply:
Ah, so the old logic was wrong and `source` couldn't actually be empty there?

Reply:
Yes, when `source` is empty, `FillConfig` sets it to `loopPath`.